packages/sdk/clients/createBicoBundlerClient.test.ts

```diff
@@ -0,0 +1,85 @@
+import { http, type Account, type Address, type Chain, isHex } from "viem"
+import type { BundlerClient } from "viem/account-abstraction"
+import { afterAll, beforeAll, describe, expect, test } from "vitest"
+import { toNetwork } from "../../test/testSetup"
+import {
+  getTestAccount,
+  killNetwork,
+  toTestClient,
+  topUp
+} from "../../test/testUtils"
+import type { MasterClient, NetworkConfig } from "../../test/testUtils"
+import contracts from "../__contracts"
+import { type NexusAccount, toNexusAccount } from "../account/toNexusAccount"
+import { createBicoBundlerClient } from "./createBicoBundlerClient"
+
+describe("bico.bundler", async () => {
+  let network: NetworkConfig
+  let chain: Chain
+  let bundlerUrl: string
+
+  // Test utils
+  let testClient: MasterClient
+  let account: Account
+  let nexusAccountAddress: Address
+  let bicoBundler: BundlerClient
+  let nexusAccount: NexusAccount
+
+  beforeAll(async () => {
+    network = await toNetwork()
+
+    chain = network.chain
+    bundlerUrl = network.bundlerUrl
+    account = getTestAccount(0)
+    testClient = toTestClient(chain, getTestAccount(5))
+
+    nexusAccount = await toNexusAccount({
+      holder: account,
+      chain,
+      transport: http()
+    })
+
+    bicoBundler = createBicoBundlerClient({ bundlerUrl, account: nexusAccount })
+    nexusAccountAddress = await nexusAccount.getCounterFactualAddress()
+    await topUp(testClient, nexusAccountAddress)
+  })
+  afterAll(async () => {
+    await killNetwork([network?.rpcPort, network?.bundlerPort])
+  })
+
+  test.concurrent("should have 4337 bundler actions", async () => {
+    const [chainId, supportedEntrypoints, preparedUserOp] = await Promise.all([
+      bicoBundler.getChainId(),
+      bicoBundler.getSupportedEntryPoints(),
+      bicoBundler.prepareUserOperation({
```

Comment:
what is prepareUseroperation method on bundler client?

Reply:
https://viem.sh/account-abstraction/actions/bundler/prepareUserOperation

Reply:
Basically fills in missing pieces of a partial userOp

Reply:
I see